File: front-end/financiamento-veiculos-study-web/src/app/shared/models/cliente-request.ts
export class ClienteRequest {
  nome: string;
  cpf: string;
  dataNascimento: string;
  telefone: string;
  renda: number;
  possuiImovel: boolean;

  constructor(nome: string, cpf: string, dataNascimento: string, telefone: string, renda: number, possuiImovel: boolean) {
    this.nome = nome;
    this.cpf = cpf;
    this.dataNascimento = dataNascimento;
    this.telefone = telefone;
    this.renda = renda;
    this.possuiImovel = possuiImovel;
  }
}

File: microservices/FinanciamentoVeiculosStudy/src/main/java/com/financiamentos/veiculosstudy/entity/endpoint/ClienteRequest.java
package com.financiamentos.veiculosstudy.entity.endpoint;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.financiamentos.veiculosstudy.serializer.IsoLocalDateDeserializer;
import com.financiamentos.veiculosstudy.serializer.IsoLocalDateSerializer;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldNameConstants;

import java.time.LocalDate;

@Getter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@FieldNameConstants
//@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
public class ClienteRequest {
    private String nome;
    private String cpf;

    @JsonSerialize(using = IsoLocalDateSerializer.class)
    @JsonDeserialize(using = IsoLocalDateDeserializer.class)
    private LocalDate dataNascimento;
    private String telefone;
    private float renda;
    private boolean possuiImovel;
}
